src/Compilers/Core/Portable/SymbolDisplay/SymbolDisplayExtensions.cs

```diff
@@ -54,6 +54,46 @@ public static string ToDisplayString(this ImmutableArray<SymbolDisplayPart> part
             }
         }
 
+        /// <summary>
+        /// Converts an ArrayBuilder of <see cref="SymbolDisplayPart"/>s to a string.
+        /// </summary>
+        /// <param name="parts">The array of parts.</param>
+        /// <returns>The concatenation of the parts into a single string.</returns>
+        [PerformanceSensitive("https://github.com/dotnet/roslyn/pull/67203", AllowImplicitBoxing = false)]
+        internal static string ToDisplayString(this ArrayBuilder<SymbolDisplayPart> parts)
+        {
+            if (parts is null)
+            {
+                throw new ArgumentException(nameof(parts));
+            }
+
+            if (parts.Count == 0)
+            {
+                return string.Empty;
+            }
+
+            if (parts.Count == 1)
+            {
+                return parts[0].ToString();
+            }
+
+            var pool = PooledStringBuilder.GetInstance();
+            try
+            {
+                var actualBuilder = pool.Builder;
+                foreach (var part in parts)
+                {
+                    actualBuilder.Append(part.ToString());
+                }
+
+                return actualBuilder.ToString();
+            }
+            finally
+            {
+                pool.Free();
```

Comment:
I thought there was a ToStringAndFree helper for this pattern.

Reply:
Yep, there is. Changed this one and the one I copied this from.